subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c

```diff
@@ -479,12 +479,22 @@ static int write_dl_uri(uint16_t obj_inst_id, uint16_t res_id, uint16_t res_inst
 	state = lwm2m_firmware_get_update_state();
 
 	if (state == STATE_IDLE) {
-		lwm2m_firmware_set_update_result(RESULT_DEFAULT);
+		lwm2m_firmware_set_update_state_inst(obj_inst_id, STATE_DOWNLOADING);
 
 		if (data_len > 0) {
 			ret = init_start_download(package_uri);
-			if (ret) {
-				lwm2m_firmware_set_update_result(RESULT_DEFAULT);
+			switch (ret) {
+			case 0:
+				/* OK */
+				break;
+			case -EINVAL:
+				lwm2m_firmware_set_update_result_inst(obj_inst_id,
+								      RESULT_INVALID_URI);
+				break;
+			case -EBUSY: /* FALLTHROUGH */
```

Comment:
Out of memory looks a bit illogical here. A comment explaining the choice of update result code wouldn't hurt.

Reply:
It was not clear which it should be. I have now set the -EBUSY to NO_STORAGE.
That error comes if download_library init fails, or if some MCUBoot buffer settings fails.